hudi-client/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -369,10 +343,56 @@ private void bootstrapFromFilesystem(JavaSparkContext jsc, HoodieTableMetaClient
       }
     });
 
-    LOG.info("Committing " + partitionFileList.size() + " partitions and " + stats[0] + " files to metadata");
+    LOG.info("Committing " + partitionToFileStatus.size() + " partitions and " + stats[0] + " files to metadata");
     update(commitMetadata, createInstantTime);
   }
 
+  /**
+   * Function to find hoodie partitions and list files in them in parallel.
+   *
+   * @param jsc
+   * @param datasetMetaClient
+   * @return Map of partition names to a list of FileStatus for all the files in the partition
+   */
+  private Map<String, List<FileStatus>> parallelFileSystemListing(JavaSparkContext jsc,
```

Comment:
- Perhaps name it `getPartitionsToFilesMapping()` ?
- Ideally, it would have been good to have it as a utility in `FsUtils` but because of our current structure we can't move it, since `FsUtils` is in `hudi-common` which we don't want to depend on `spark`.

Reply:
Done